### PR TITLE
Add reactive variable for update method call

### PR DESCRIPTION
**Pull request by ebenezer-melkamu**
## Changelog 
 1. Added a reactive variable to show loading state 
 2. Added a README file

### CODE REVIEW

File: imports/ui/Task.html [hunk 1]
```diff
@@ -2,6 +2,9 @@
   <li>
     <input type="checkbox" class="toggle-checked" checked="{{isChecked}}" />
     <span>{{text}}</span>
+    {{#if isUpdateLoading}}
+    <div class="loading">loading...</div>
```
Comment (dr-dimitru) [inline imports/ui/Task.html]: @AbegaM indentation

File: imports/ui/Task.js [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
 import "./Task.html";
 
+Template.task.onCreated(function () {
+  this.isUpdateLoading = new ReactiveVar(false);
+});
+
 Template.task.events({
   "click .toggle-checked"(event) {
+    Template.instance().isUpdateLoading.set(true);
     Meteor.call("tasks.setIsChecked", this._id, !this.isChecked);
```
Comment (dr-dimitru) [inline imports/ui/Task.js]: This is callback function, so method call won't wait to end to call `Template.instance().isUpdateLoading.set(false);`

@AbegaM please change this call to `async/await` or callback type. Do not forget to check and log for errors if any returned from the server

**Review by dr-dimitru — CHANGES_REQUESTED** [review]
@AbegaM please pardon me for the late reply/review, we had too many applicants to review for. You did a good job and I'd like to give you a little bigger scope and real task.

I hope you're still available, if so please:
1. Fix all comments/issues in this PR
2. Merge PR
3. Book a call using the same link for on-boarding

File: imports/ui/Task.html (hunk 1 above)
Comment (dr-dimitru) [reply imports/ui/Task.html]: @AbegaM add styling if necessary as well

Comment (ebenezer-melkamu) [reply imports/ui/Task.html]: I have fixed the indentation and have added a simple animation for the spinner

File: imports/ui/Task.js (hunk 2 above)
Comment (ebenezer-melkamu) [reply imports/ui/Task.js]: I have added a `try catch` function to catch and log errors that are coming from the backend server
 ```js
   try {
       Template.instance().isUpdateLoading.set(true);
       await Meteor.call("tasks.setIsChecked", this._id, !this.isChecked);
       Template.instance().isUpdateLoading.set(false);
     } catch (error) {
       console.log(error);
    }
 ```

**Comment by dr-dimitru** [comment]
@AbegaM I had more comments would you open a new PR?
this line needs a fix:
```
await Meteor.call("tasks.setIsChecked", this._id, !this.isChecked);
```

https://docs.meteor.com/api/methods.html

**Comment by dr-dimitru** [comment]
@AbegaM 

0. Fix `await Meteor.call("tasks.setIsChecked", this._id, !this.isChecked);`
1. Call all methods as async/await
2. Call `setTimeout`around method as described in my comments
3. Call `Template.instance()` only once per each helper

**Comment by ebenezer-melkamu** [comment]
> 

@dr-dimitru For the first comment, do you expect me to make an Asynchronous call to the server like this?
  ```js
      Meteor.call('foo', 1, 2, (error, result) => { ... });
   ```

**Comment by dr-dimitru** [comment]
@AbegaM please see the docs, it's all in there, look for await/async Meteor method call — 

https://docs.meteor.com/api/methods.html (same link as above)

**Comment by ebenezer-melkamu** [comment]
@dr-dimitru Here is a new pull request, please review it and let me know what you think 
 https://github.com/AbegaM/meteor-blaze-example/pull/2

**Pull request merged**